pkg/logevents/logevents.go

```diff
@@ -0,0 +1,87 @@
+/*
+* Copyright 2020 Armory, Inc.
+
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+
+*    http://www.apache.org/licenses/LICENSE-2.0
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package logevents
+
+import (
+	"encoding/json"
+	"github.com/armory/dinghy/pkg/cache"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+type LogEventsClient interface {
+	GetLogEvents() ([]LogEvent, error)
+	SaveLogEvent(logEvent LogEvent) error
+}
+
+type LogEventRedisClient struct {
+	RedisClient *cache.RedisCache
+}
+
+type LogEvent struct {
+	Org 		string
+	Repo 		string
+	Files		[]string
+	Message		string
+	Date		int64
+	Commits		[]string
+}
+
+
+
+func (c LogEventRedisClient) GetLogEvents() ([]LogEvent, error) {
+	loge := log.WithFields(log.Fields{"func": "GetLogEvents"})
+	keys, _, err := c.RedisClient.Client.Scan(0, cache.CompileKey("logEvent*"), 1000).Result()
+	if err != nil {
+		loge.WithFields(log.Fields{"operation": "scan key", "key": cache.CompileKey("logEvent*")}).Error(err)
+		return nil, err
+	}
+	var result []LogEvent
+	for _, key := range keys {
+		currentEventLog, errorNoKey := c.RedisClient.Client.Get(key).Result()
+		if errorNoKey != nil {
+			loge.WithFields(log.Fields{"operation": "get key", "key": key}).Error(err)
+			continue
+		}
+		var logEvent LogEvent
+		errorUnmarshal := json.Unmarshal([]byte(currentEventLog), &logEvent)
+		if errorUnmarshal != nil {
+			loge.WithFields(log.Fields{"operation": "unmarshall key " + key, "content": currentEventLog}).Error(err)
+			continue
+		}
+		result = append(result, logEvent)
+	}
+	return result, nil
+}
+
+func (c LogEventRedisClient) SaveLogEvent(logEvent LogEvent) error {
+	loge := log.WithFields(log.Fields{"func": "SaveLogEvent"})
+	nanos := time.Now().UnixNano()
+	milis := nanos / 1000000
+	logEvent.Date = milis
+	logEventBytes, err := json.Marshal(logEvent)
+	if err != nil {
+		loge.WithFields(log.Fields{"operation": "marshall logEvent", "content": logEvent}).Error(err)
+		return err
+	}
+	key := cache.CompileKey("logEvent" + string(milis))
+	if _, err := c.RedisClient.Client.Set(key, logEventBytes, 1 * time.Hour).Result(); err != nil {
```

Comment:
Will this set the TTL to 1 hour? What about setting this to a Day instead?

Reply:
It was for testing, now it will be 1440 minutes by default, but it can be configurable in the dinghy.yml